src/server/game/AI/CreatureAI.cpp

```diff
@@ -382,6 +382,19 @@ void CreatureAI::SetBoundary(CreatureBoundary const* boundary, bool negateBounda
     me->DoImmediateBoundaryCheck();
 }
 
+void CreatureAI::CheckMeleeRepositionRequirements()
+{
+    if (Unit* victim = me->GetVictim())
+    {
+        Position victimPos;
+        victimPos.Relocate(victim->GetPosition());
+
+        // If we are closer than 50% of the combat reach we are going to reposition ourself
+        if (me->GetDistance(victimPos) < CalculatePct(me->GetCombatReach(), 50))
```

Comment:
me->GetDistance(victimPos) includes me->GetCombatReach(), that means if your exact distance is less than combat reach, it will return 0.0f

Reply:
Huh? I see      float d = GetExactDist(&pos) - GetObjectSize(); in that function. So where exactly is the combat reach getting taken into account so I can adapt the formular?

Reply:
see how GetObjectSize() is implemented

Reply:
Ye. Did so. I see no problem there.  It was intended to get the distance to be unaffected by combat reach because I wanted to make sure that we take the actual distance rather than checking when the target enters my combat reach. The backwards movement it not suposed to make the target move to the very outer circle radius just like the initial chase movement does. On retail they just move a bit back. This is also why I don't use any range values in the next step to get the contact point.

Reply:
I don't see a problem there. It works exactly how I wanted it to work: compare distance between the two positions regardless of their combat-reach, if the victim is too close the creature will step back. If I would be a big dragon and a small player attacks me and moves closer it would trigger my repositioning when he comes closer than 50% of my combat-reach as actual distance between those two positions. In the next step I use the ContactPoint helper to get a good distance and so the dragon will move back.

Since the attack range is calculated by using  AttackerCombatreach + VictimCombatreach there is no issue with that because the reposition will place you far enough to prevent a double triggering.

Reply:
in case of your last picture (bottom right):
big combat reach: 10
small combat reach: 1
-> exact dist: 11
-> distance `float d = GetExactDist(&pos) - GetObjectSize()`: 11 - 10 = 1
-> 1 < 10 * 50%
=> repositioning
or I'm wrong?

Reply:
It is more like:
-> distance between point 1 and point 2 (ignoring combatreach entirely)
-> distance < AttackerCombatreach * 50%
--> if yes: reposition

It's a bit tricky how I applied that handling. I do not compare the combat-reach distance from those creatures. I use the attacker combatreach only to know when the creatures comes too close. Not by using combatreach, more by using exact positions.